sdk/ml/azure-ai-ml/azure/ai/ml/dsl/_fl_scatter_gather_node.py

```diff
@@ -0,0 +1,131 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+# pylint: disable=protected-access
+from typing import List, Dict, Optional, Union
+import importlib
+
+from azure.ai.ml.entities import PipelineJob
+from azure.ai.ml.entities._builders.fl_scatter_gather import FLScatterGather
+from azure.ai.ml.entities._assets.federated_learning_silo import FederatedLearningSilo
+from azure.ai.ml.entities import CommandComponent
+from azure.ai.ml._utils._experimental import experimental
+
+
+def _check_for_import(package_name):
+    try:
+        # pylint: disable=unused-import
+        importlib.import_module(package_name)
+    except ImportError:
+        raise ImportError(
+            "The DSL FL Node has an additional requirement above the rest of the "
+            + "AML SDK repo in that the mldesigner package is required. Please run `pip install mldesigner` "
+            + "and try again."
+        )
+
+
+@experimental
+def fl_scatter_gather(
+    *,
+    silo_configs: List[FederatedLearningSilo],
+    silo_component: Union[PipelineJob, CommandComponent],
+    aggregation_component: Union[PipelineJob, CommandComponent],
```

Comment:
For `silo_component` and `aggregation_component`, I understand `PipelineComponent` and `CommandComponent` shall be more proper for annotation?

Reply:
Based on the typing that we check in the anchoring file, and based on my understanding of pipeline object structure, a PipelineJob should always be wrapping a PipelineComponent, and we're more interested in the former, so I don't think this needs to change.